helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -901,6 +904,11 @@ public void addPreConnectCallback(PreConnectCallback callback) {
 
   @Override
   public boolean isLeader() {
+    return isInstanceLeader(null);
+  }
+
+  @Override
+  public boolean isInstanceLeader(InstanceLeaderSession instanceLeaderSession) {
```

Comment:
As we discussed,
1. make it private if not used anywhere else.
2. make it getSessionIdIfLeader() which returns an Optional object if the node is leader.